sync/sync.go

```diff
@@ -184,7 +184,10 @@ func (s *Synchronizer) verifierTask(ctx context.Context, block *core.Block, stat
 				}
 				resetStreams()
 				return
+			} else {
```

Comment:
you dont need the else statement here. `if err != nil` branch returns early anyways.